### ISSUE TITLE
How to visualize the electric or magnetic fields in DD4hep

### ISSUE THREAD

**Issue by TrT-TOT** [AIDASoft/DD4hep#1203]
Hi all,
  I'm trying to constrcut a set of magnetic fields in one geometry. I'd like to visualize the setup of field to make sure everyhing is okay. Is it possible via geoDisplay or some other tools? Thanks!

Best regards,
   Zekun

**Comment by atolosadelgado** [comment]
Dear Zekun,

Geant4 provide a dedicated [UI command](https://geant4-userdoc.web.cern.ch/UsersGuides/ForApplicationDeveloper/html/Control/AllResources/Control/UIcommands/_vis_scene_add_.html#c17) to add the magnetic field visualization to a current scene. The Qt GUI of Geant4 can be used via `ddsim` with flag `--runType qt` , see [this tutorial](https://hep-fcc.github.io/fcc-tutorials/master/full-detector-simulations/Visualization/Visualization.html#geant4).


To open a Qt window and draw the whole geometry you need to provide some UI commands, which can be placed inside a macro file and feeded to ddsim with flag `--macroFile vis.mac`

```
# vis.mac
# Use this open statement to create an OpenGL view:
/vis/open OGL 600x600-0+0

# Draw geometry:
/vis/drawVolume

# Draw smooth trajectories at end of event, showing trajectory points
# as markers 2 pixels wide:
/vis/scene/add/trajectories smooth

# Draw hits at end of event:
/vis/scene/add/hits

# To superimpose all of the events from a given run:
/vis/scene/endOfEventAction accumulate

# To get nice view
# Make the "world_volume" box invisible
/vis/geometry/set/visibility world_volume 0 false

# Plot magnetic field
/vis/scene/add/magneticField

```

To sum up, you may use Geant4 Qt for visualization, via ddsim with the following command,
`ddsim --compactFile ./compact/mydetector.xml --runType qt --macroFile vis.mac`

I hope this helps :)

Best,
Alvaro

**Comment by TrT-TOT** [comment]
Dear Alvaro and Markus,
  Thanks for the advice! I think that's enough for my purpose! Thanks again.
